### ISSUE TITLE
MLP writes raw logits into the class-probability columns (and uncertainty is computed from them)

### ISSUE THREAD

**Issue by felixbur** [felixbur/nkululeko#446]
_Found with nkululeko 1.11.0; code location checked against master 15f2ad4._

**File:** `nkululeko/models/model_mlp.py`, `get_probas`

```python
def get_probas(self, logits):
    # make a dataframe for probabilites (logits)
    ...
    for i, c in enumerate(classes):
        proba_d[c] = list(logits.numpy().T[i])
```

No softmax is applied, so the per-class "probability" columns in the saved prediction CSVs (`results/*_test_0_*.csv` etc.), and anything derived from them (e.g. the `uncertainty` column and uncertainty plots), are based on raw logits. The `predicted` label is still correct, because argmax is unchanged.

**Observed:** WavLM-embedding MLP, binary asthma task. Values in the `no` / `yes` columns are negative or above 1 (mean of `yes` on test: −0.84). Thresholding `yes >= 0.5` gives UAR 0.620 instead of the reported 0.712. Any downstream use of the probabilities (session averaging, calibration, ROC with a probability threshold) is silently wrong.

**Fix:**

```python
probs = torch.softmax(logits, dim=1).numpy()
for i, c in enumerate(classes):
    proba_d[c] = list(probs.T[i])
```

**Comment by felixbur** [comment]
fixed by 1.11.4
